cloud/amazon/ec2_asg_facts.py

```diff
@@ -0,0 +1,369 @@
+                "enabled_metrics": [],
+                "health_check_grace_period": 300,
+                "health_check_type": "ELB",
+                "instances":
+                [
+                    {
+                        "availability_zone": "us-west-2c",
+                        "health_status": "Healthy",
+                        "instance_id": "i-047a12cb",
+                        "launch_configuration_name": "public-webapp-production-1",
+                        "lifecycle_state": "InService",
+                        "protected_from_scale_in": false
+                    },
+                    {
+                        "availability_zone": "us-west-2a",
+                        "health_status": "Healthy",
+                        "instance_id": "i-7a29df2c",
+                        "launch_configuration_name": "public-webapp-production-1",
+                        "lifecycle_state": "InService",
+                        "protected_from_scale_in": false
+                    }
+                ],
+                "launch_configuration_name": "public-webapp-production-1",
+                "load_balancer_names": ["public-webapp-production-lb"],
+                "max_size": 4,
+                "min_size": 2,
+                "new_instances_protected_from_scale_in": false,
+                "placement_group": None,
+                "status": None,
+                "suspended_processes": [],
+                "tags":
+                [
+                    {
+                        "key": "Name",
+                        "propagate_at_launch": true,
+                        "resource_id": "public-webapp-production",
+                        "resource_type": "auto-scaling-group",
+                        "value": "public-webapp-production"
+                    },
+                    {
+                        "key": "env",
+                        "propagate_at_launch": true,
+                        "resource_id": "public-webapp-production",
+                        "resource_type": "auto-scaling-group",
+                        "value": "production"
+                    }
+                ],
+                "termination_policies":
+                [
+                    "Default"
+                ],
+                "vpc_zone_identifier":
+                [
+                    "subnet-a1b1c1d1",
+                    "subnet-a2b2c2d2",
+                    "subnet-a3b3c3d3"
+                ]
+            }
+        ]
+    """
+    asgs = conn.describe_auto_scaling_groups()['AutoScalingGroups']
+    matched_asgs = []
+    for asg in asgs:
+        if name:
+            name_prog = re.compile(r'^' + name)
+            matched_name = name_prog.search(asg['AutoScalingGroupName'])
+        else: matched_name = True
+
+        if tags: matched_tags = match_asg_tags(tags, asg)
+        else: matched_tags = True
+
+        if matched_name and matched_tags:
+            asg_details = dict(created_time=asg['CreatedTime'].isoformat())
```

Comment:
Assuming speedy merge of https://github.com/ansible/ansible/pull/15200, you can replace everything in this if block with:

`asg['CreatedTime'] = asg['CreatedTime'].isoformat()
asg_details = camel_dict_to_snake_dict(asg)`